### DISCUSSION TITLE
public array property transform from Carbon to String between first call and action call.

### DISCUSSION THREAD

**Discussion by sebreunion** [livewire/livewire#4390] [1. Help] [answered]
Hi,

In my component, i have a public property : 

`    public $per;
`

In the mount method, i set this property :

`$per = CarbonPeriod::create($this->cli->first, '1 month', 12);`
`$this->per = $per->toArray();`

When i first load the page, i can dd $per in the blade and i got a nice array of 12 Illuminate\Support\Carbon, so that i can ->format('d/m/Y') them.

Anytime an action is called, my array is turned into an array of 12 Strings and i cannot ->format() them anymore without getting an error.

The only solution found is to re-set this public property anytime in my render() method before rendering the blade.

I would like to set this property in my mount method only once to prevent database query and useless code.

It's not the first time that i encounter such behavior of property slightly changing between the first call and action call. 

To solve this, i always move my code from mount() to render() even if those property don't need to be set again each call as they don't change.

Could you please help me sort this out ?

Thx

**Comment by joshhanley** [comment, marked as answer]
@sebreunion Livewire doesn't know how to serialise an array of carbon instances when it sends them to the front end, that's why you will be seeing date strings on the next request.

I would probably create a computed property instead for your date range. https://laravel-livewire.com/docs/2.x/properties#computed-properties

Hope this helps!
